resources/design-document.md

```diff
@@ -1,75 +1,197 @@
 
-_Describe the behavior of the first endpoint you will build into your service API. This should include what data it requires, what data it returns, and how it will handle any known failure cases. You should also include a sequence diagram showing how a user interaction goes from user to website to service to database, and back. This first endpoint can serve as a template for subsequent endpoints. (If there is a significant difference on a subsequent endpoint, review that with your team before building it!)_
+This will be accomplished using a RESTful API and AWS Lambda to create endpoints.
 
-_(You should have a separate section for each of the endpoints you are expecting to build...)_
+Inventory item and warehouse data will be stored in DynamoDB
+
+PopStock will also provide a web interface for users to manage their warehouses and inventory.  There will be a main page showing existing warehouses, with buttons to create a new one or modify existing inventory items. The inventory item page will have buttons linking to pages to create, update, or delete existing inventory items. Clicking the button for an existing warehouse will lead to a page where you can update a warehouse's inventory, run reports, or generate a new layout.  
+# 6. API
+
+## 6.1. Public Models
+```
+//WarehouseModel 
+String itemId;
+String name;
+int region;
+Map<ItemModel, Integer> inventory
+int size
+```
+
+```
+//ItemModel
+String warehouseId;
+String name;
+int regionalDemand;
+double salesForecast
+double perPallet;
+double weight;
+double purchaseCost;
+double baseMargin;
+double rateOfReplenishment;
+String category
+String synergy
+boolean active
+```
+
+```
+//shippedModel
+String itemId // partition key, string
+String warehouseId, sort key, string
+Date date
+int count
+String customerId
+```
+
+## 6.2. Create Item Endpoint
+
+* Accepts POST requests to /items
+* Accepts input data and user info to create a new item and returns an itemModel for the logged-in user
+* 
+## 6.3 Get Item Endpoint
+* Accepts GET requests to /items/id
+* Accepts an itemId and user info and returns the relevant itemModel
+
+## 6.4 Remove Item Endpoint
+* Accepts PUT requests to /items/id/remove
+* Accepts an itemID, sets active to false for the given item, and returns the updated itemModel for the 
+
+## 6.5 Update Item Endpoint
+* Accepts PUT requests to /items/id
+* Accepts an itemId, input data, and user info, updates the item data, and returns an itemModel
+
+## 6.6 Create Warehouse Endpoint
+* Accepts POST requests to /warehouses
+* Accepts input data and user info to create a new warehouse
+* Retrieves a list of all items and calculates the most profitable selection for the given warehouse metrics
+* Returns the created warehouseModel
```

Comment:
Good catch, I'll also need this logic separated anyway for inventory update suggestions

Reply:
So the calculation logic itself will need to be separated from the activity of this endpoint, so that it can also be used to generate inventory change suggestions in the inventory report, however the idea with the app is that with a small amount of user input and a button click, a full warehouse inventory list is generated based on "third-party data" (simulated with manual item data input)